File: src/sports_management/modules/booking_module/lib/domain/usecases/get_booking_detail_usecase.dart
import 'package:server_module/server_module.dart';
import '../entities/booking_detail_entity.dart';

class GetBookingDetailUseCase {
  final BookingRepository _repository;

  GetBookingDetailUseCase(this._repository);

  Future<BaseResponse<BookingDetailEntity>> call(String id) async {
    final response = await _repository.getBookingById(id);

    if (response.success && response.data != null) {
      final data = response.data;
      if (data is BookingDetailEntity) {
        return BaseResponse(
          success: true,
          message: response.message,
          data: data,
        );
      }
    }

    return BaseResponse(
      success: response.success,
      message: response.message,
      data: null,
    );
  }
}

File: src/react-staff-admin/src/features/booking/domain/usecases/get_booking_detail.usecase.ts
import { BookingRepository } from '../repositories/booking.repository';
import { Booking } from '../entities/booking.entity';

export class GetBookingDetailUseCase {
  constructor(private repository: BookingRepository) {}

  async execute(id: string): Promise<Booking | null> {
    return this.repository.getBookingById(id);
  }
}
